pg_shard.c

```diff
@@ -703,8 +739,8 @@ RowAndColumnFilterQuery(Query *query)
 	filterQuery = makeNode(Query);
 	filterQuery->commandType = CMD_SELECT;
 	filterQuery->rtable = rangeTableList;
-	filterQuery->targetList = targetList;
 	filterQuery->jointree = fromExpr;
+	filterQuery->targetList = targetList;
```

Comment:
Yes. The above blocks build `rangeTableList` then `fromExpr` then `targetList`. I felt the assignments should mirror the order of construction.